src/examples/s3-storage/src/backend/custom-mutations/index.ts

```diff
@@ -0,0 +1,86 @@
+import {
+	Field,
+	fromBackendEntity,
+	graphweaverMetadata,
+	ID,
+	InputType,
+	ResolverOptions,
+} from '@exogee/graphweaver';
+import { ConnectionManager } from '@exogee/graphweaver-mikroorm';
+import { Jimp } from 'jimp';
+import { pgConnection } from '../database';
+import { s3 } from '../s3';
+import { Submission } from '../schema';
+@InputType('CreateThumbnailInput')
+class CreateThumbnailInput {
+	@Field(() => ID)
+	submissionId: number;
+
+	@Field(() => Number)
+	width: number;
+
+	@Field(() => Number)
+	height: number;
+}
+
+graphweaverMetadata.addMutation({
+	name: 'createThumbnail',
+	getType: () => Submission,
+	args: { input: () => CreateThumbnailInput },
+	resolver: async ({
+		args,
+		context,
+		source,
+		info,
+		fields,
+	}: ResolverOptions<{ input: CreateThumbnailInput }>) => {
+		// get the metadata of the submission to copy
+		const database = ConnectionManager.database(pgConnection.connectionManagerId);
+		const submission = await database.em.findOneOrFail(Submission, {
+			id: args.input.submissionId.toString(),
+		});
+
+		const filename = submission.image.filename;
+
+		if (!filename) {
+			throw new Error('No filename attached to submission');
+		}
+
+		// fetch the image data
+		const imageUrl = await s3.getDownloadUrlForKey(filename);
+
+		// resize the image to the desired dimensions
+		const input = await Jimp.read(imageUrl);
+		const resizedImage = await input
+			.resize({ w: args.input.width, h: args.input.height })
+			.getBuffer('image/png');
+
+		const upload = await s3.getUploadUrl({
+			args: { key: filename },
+			source,
+			context,
+			info,
+			fields,
+		});
+
+		// upload the image to s3
+		await fetch(upload.url, {
```

Comment:
I think we'll need to get a signed URL here for the upload or we'll get a 403 on AWS. In Minio this probably works, but I doubt it would in prod.

Reply:
Thankfully whoever implemented getUploadUrl originally in the s3 provider already thought about that 😅

Reply:
Nice.